packages/core/src/components/chip/chip.tsx

```diff
@@ -55,7 +55,16 @@ export class TdsChip {
   }>;
 
   private handleChange = () => {
-    this.checked = !this.checked;
+    if (this.type === 'checkbox') {
+      // Toggle the prop on click
+      this.checked = !this.checked;
+    } else if (this.type === 'radio') {
+      // Always set it to true to enforce visual update for selected state
+      this.checked = true;
+    } else {
+      console.warn('Unsupported type in Chip component!');
+    }
+
```

Comment:
@timrombergjakobsson, I have been looking into this one for longer and found a bug that causes this.

The type button and type checkbox do not have this issue as they can be used as a single component. On the other side, the radio button is a component where you use at least two instances of it. During debugging, I was looking at what happens with the "checked" attribute in the DOM. Once it is assigned to the Chip that is radio type it stays there forever. Clicking on another Chip of the same radio type selects that one, but the previous one still has a "checked" attribute. Looking into native HTML input with type radio and other Design systems we realized the same happens there. We have tested it in form too, so form, on submitting will pick up "last" Chip on which user has clicked - no matter all of them might have "checked" as attribute.

What are possible options:

1. Add @Listen decorator to Chip component so it listens to its own tdsChange. If that specific instance of Chip component is emitting, it will set its own checked prop to true, while all the others will set it to false. In this example, we would need to enable Chip components with same "name" prop to be able to communicate to each other or better say - be aware of each other.
2. On click of Chip component, set checked prop to true (even if it already is true) to enforce styling update to blue. 

Number 2. you already did so kudos for that, I just reverted some changes to keep changes smaller. Please review it too and ping me if something more is needed.

This was indeed very interesting issue to debug. 

PS: Nevermind for multiple console logs on tdsChange, it seems to be issue related to Storybook, Chrome. I have checked it in Angular demo and there seems fine. So sorry to bother you around it. I noticed different behaviors of local environment if Chome and Firefox as well as in incognito mode.

Reply:
@theJohnnyMe wow, interesting little bug we discovered here. So I reviewed your changes, and they seem to work. However, the local behavior of changeEvents makes it hard to trust that they are working properly with your changes. Did you `npm pack`and tried it out locally in Angular env to confirm that it worked?  If so, I'm ok with the changes you made, as it visually seems to work as intended.

Reply:
@timrombergjakobsson  - I agree with you, I have not tested in Angular, but we should do it.
 It might be a bit tricky but not impossible:

- We would need to pack the core
- Then put it in angular and pack that one
- Test it in Angular demo

Let's try that during the day and see how it goes